File: Sliding Window/0713. Subarray Product Less Than K/0713.py
class Solution:
    def numSubarrayProductLessThanK(self, nums: List[int], k: int) -> int:
        if k<=1:
            return 0
        
        l=res=0
        curr=1

        for r in range(len(nums)):
            curr*=nums[r]
            while curr>=k:
                curr//=nums[l]
                l+=1
            res+=(r-l+1)
        
        return res

File: Sliding Window/0713. Subarray Product Less Than K/0713.cpp
class Solution {
public:
    int numSubarrayProductLessThanK(vector<int>& nums, int k) {
        ios_base::sync_with_stdio(0);
        cin.tie(0);
        cout.tie(0);

        if (k<=1) {
            return 0;
        }

        int curr=1;
        int l=0;
        int res=0;

        for(int r=0; r<nums.size(); ++r) {
            curr*=nums[r];
            while(curr>=k && l<=r) {
                curr/=nums[l];
                l++;
            }
            res+=(r-l+1);
        }
        return res;

    }
};